### ISSUE TITLE
Increase batch sizes in examples/benchmarks

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#468]
Most of the examples use very small batch sizes, which leads to a lot of tiny GPU kernels and poor GPU utilization.  Increasing it can help a lot.  For example, on the Tox21 example I find that increasing the batch size from 50 to 1000 cuts the training time in half.  We should go through all the examples and benchmarks and consider what batch size to use for each one.

**Comment by patrickhop** [comment]
When you say training time, is that to reach convergence or complete N epochs? I would have expected the smaller mini batch sizes to reach convergence in less time?

**Comment by evanfeinberg** [comment]
Thank you @peastman, this is great! Faster training means broader
hyperparameter searches and better models :)

@Patrick Hop thanks for chiming in. I don't think there's a monotonic
negative relationship between batch size and convergence time. If anything
the batch size itself is considered a hyperparam to search over that seems
largely system-dependent (cf. https://arxiv.org/pdf/1206.5533.pdf)

**Comment by patrickhop** [comment]
@evanfeinberg It's definitely hard to say anything about monotonicity in general here. For example, for a fixed learning rate, the update need not be contractive for all batch sizes - it need not converge at all.

From the paper you linked to: "The mini-batch size (B in Eq. (1)) is typically chosen between 1 and a few hundreds, e.g. B = 32 is a good default value, with values above 10 taking advantage of the speed-up of matrixmatrix products over matrix-vector products."

Below is a newer paper that you might like. Larger batch sizes seem to considerably restrict the range of acceptable hyperparameters. This suggests that if we want to select a default batch size, we should probably pick one on the smaller side.
https://arxiv.org/pdf/1508.02788.pdf

Another consideration is smaller datasets. For the delaney dataset, a batch size of 1000 brings us up to nearly full blown gradient descent.

**Comment by evanfeinberg** [comment]
Great points Patrick, thanks for sharing. Seems to be a similar conclusion
to this paper (https://arxiv.org/pdf/1609.04836.pdf) where larger batch
sizes lead to steeper minima which is less desirable for generalization to
test.

**Comment by peastman** [comment]
This is the `tox21_tf_models.py` example, which runs a fixed number of epochs.  The accuracy at the end is not significantly affected by the change.  I've also tested a few others and seen similar results.

I agree that there's not a single "best" batch size we can use for all models.  We need to evaluate this separately for each one.  But the little bit of testing I've done so far suggests that most of them are setting the batch size too low, and increasing it can speed them up without hurting the final result.

**Comment by rbharath** [comment]
Perhaps a good solution may be to increase batch-size on one example at a time, while verifying that benchmark numbers don't change more than 1% relative to the the existing numbers?

**Comment by rbharath** [comment]
I think most of these optimizations have been finished, so closing issue. Feel free to re-open if more improvements can be made.
